specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/Alert.json

```diff
@@ -0,0 +1,36 @@
+{
+    "parameters": {
+      "api-version": "2018-08-01-preview",
+      "subscriptionId": "01234567-89ab-cdef-0123-456789abcdef",
+      "resourceGroupName": "rg1",
+      "alertId": "fae17478-31e7-40fd-b62a-1a127754a463_0e1d6ee7-0052-4996-89e2-023263106885"
+    },
+    "responses": {
+      "200": {
+        "body": {
+          "id": "/subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourceGroups/rg1/providers/Microsoft.CostManagement/alerts/fae17478-31e7-40fd-b62a-1a127754a463_0e1d6ee7-0052-4996-89e2-023263106885",
+          "name": "fae17478-31e7-40fd-b62a-1a127754a463_0e1d6ee7-0052-4996-89e2-023263106885",
```

Comment:
AlertId looks like it is 2 guids with an underscore between them, why are these formulated in this way? This seems very difficult for a customer to use e.g. if you were to have a PS or CLI cmd to get the details of a specific alert I would have to pas it anargument of  -Name fae17478-31e7-40fd-b62a-1a127754a463_0e1d6ee7-0052-4996-89e2-023263106885. Can this be simplified?

Reply:
Also how do alerts get created?

Reply:
> AlertId looks like it is 2 guids with an underscore between them, why are these formulated in this way? This seems very difficult for a customer to use e.g. if you were to have a PS or CLI cmd to get the details of a specific alert I would have to pas it anargument of -Name fae17478-31e7-40fd-b62a-1a127754a463_0e1d6ee7-0052-4996-89e2-023263106885. Can this be simplified?

You are right. After discussing this with my team, we can indeed simplify the alertId/name to be one guid (the second one) instead of two (the first one was meant to be the scope ID, but we managed to get it in another way).
Thanks for you suggestion!



> Also how do alerts get created?

Currently the Alert API does not allow creation. Alerts are created through a processor according to rules we defined (for example, when a budget-notification is created).
In the future we will add an API for creating alerts.

Reply:
is it possible to use something  other than a guid to name these? Using a Guid makes user interaction really difficult listing or searching in the portal is hard , as I noted before using cli and powrhsell is cumbersome, I understand from what you said that these are generated from rules, is there a way to use the rulename as a prefix for the alert or something along those lines, happy to jump on a call and disucss this

Reply:
We use Guid's to be able to identify alerts uniquely.
Quoting my team manager (email discussion):

> We need a way to distinct multiple alerts with same type and maybe with same properties.
AlertId here behave as SubscriptionId,  Guid seems to the most natural way to distinguish between alerts.
Broadening discussion, most Azure resources (application, subscription, and many more....) use Guid to distinguish between instances and that’s what we need here.